src/subscriptionNotifier/subscriptionNotifier.service.ts

```diff
@@ -0,0 +1,237 @@
+import { Injectable, Logger } from '@nestjs/common';
+import * as cron from 'node-cron';
+import { ConfigService } from '@nestjs/config';
+import { DiscordService } from 'src/discord/discord.service';
+import { SubscriptionService } from 'src/subscription/subscription.service';
+import { Subscriber } from 'prisma/prisma.types';
+import { activities, cabins, subscriptions } from '@prisma/client';
+import { ActivityService } from 'src/activity/activity.service';
+import { CabinService } from 'src/cabin/cabin.service';
+import { EmbedBuilder } from 'discord.js';
+
+type MessageThread = {
+  topic: string | null;
+  mainMessage: string;
+  embeds: EmbedBuilder[];
+};
+
+@Injectable()
+export class SubscriptionNotifierService {
+  private readonly logger = new Logger(SubscriptionNotifierService.name);
+  private task: cron.ScheduledTask;
+
+  constructor(
+    private readonly activityService: ActivityService,
+    private readonly cabinService: CabinService,
+    private readonly configService: ConfigService,
+    private readonly discordService: DiscordService,
+    private readonly subscriptionService: SubscriptionService,
+  ) {
+    if (this.configService.get('NODE_ENV') === 'production') {
+      this.task = cron.schedule('0 16 * * *', this.on4pm);
+      this.logger.log('setup 4pm subscription notifier');
+    }
+  }
+
+  on4pm = () => {
+    this.logger.log('on4pm');
+    this.notifySubscribers()
+      .catch((error) => {
+        this.logger.error(
+          'Sending subscription notifications failed with error.',
+          error,
+        );
+      })
+      .then(() =>
+        this.logger.log('Subscription notifications sent successfully.'),
+      );
```

Comment:
```suggestion
      .then(() =>
        this.logger.log('Subscription notifications sent successfully.'),
      )
      .catch((error) => {
        this.logger.error(
          'Sending subscription notifications failed with error.',
          error,
        );
      });
      
```
.then chained on to a .catch will always run.

Reply:
Makes sense, will fix that!